mmengine/fileio/backends/aws_backend.py

```diff
@@ -0,0 +1,387 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import io
+import os
+import re
+import tempfile
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Generator, Iterator, Optional, Tuple, Union
+
+from .base import BaseStorageBackend
+
+
+class AWSBackend(BaseStorageBackend):
+    """AWS storage bachend.
+
+    AWSBackend supports reading and writing data to aws
+    It relies on awscli and boto3, you must install and run ``aws configure``
+    in advance to use it.
+
+    Args:
+        path_mapping (dict, optional): Path mapping dict from local path to
+            Petrel path. When ``path_mapping={'src': 'dst'}``, ``src`` in
+            ``filepath`` will be replaced by ``dst``. Default: None.
+
+    Examples:
+        >>> filepath = 'aws://bucket/obj'
+        >>> backend = AWSBackend()
+        >>> backend.get(filepath1)  # get data from aws
+        b'hello world'
+    """
+
+    def __init__(self, path_mapping: Optional[dict] = None):
+        try:
+            import boto3
+            from botocore.exceptions import ClientError
+        except ImportError:
+            raise ImportError('Please install boto3 to enable '
+                              'AWS_Backend.')
+
+        self._client = boto3.client('s3')
```

Comment:
The [client](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.client) accepts many parameters. Should we also support them?

```python
class AWSBackend(BaseStorageBackend):
    def __init__(
        self,
        aws_access_key_id: Optional[[str] = None,
        aws_secret_access_key: Optional[[str] = None,
        path_mapping: Optional[dict] = None,
        client_kwargs: Optional[dict] = None,
    )  
        if client_kwargs is not None:
            client_kwargs = dict()
        self._client = boto3.client('s3', aws_access_key_id=aws_access_key_id, aws_secret_access_key=aws_secret_access_key, **client_kwargs)
```